### ISSUE TITLE
eslint-plugin-tailwindcss does not work properly with extension

### ISSUE THREAD

**Issue by Nurai1** [microsoft/vscode-eslint#1706]
Extension can not resolve config from eslint settings (tailwindcss settings property) while using cli execute this plugin in correct way. 

What should you do to see this issue:
1. Extend some variable to tailwind theme, for example primary-100 color.
2. Then add it ti your code.
3. Add eslint-plugin-tailwindcss to your eslint configuration.
4. Check that vs code extension see bg-primary-100/text-primary-100 as arbitrary classes instead of classes from tailwind theme.
5. Execute cli by npx eslint, no such problem

Below there are screenshots with "vs code ext" incorrect warning. primary-800 is my color from tailwind config.
<img width="896" alt="image" src="https://github.com/microsoft/vscode-eslint/assets/45640899/4ab32f85-0350-40de-adf2-37193d67d7bb">

**Comment by dbaeumer** [comment]
@Nurai1 can you please provide me with a minimal GitHub repository I can clone that demos this. This ensures that I use exactly the same setup you are seeing the problem in.

**Comment by joel-saupe-visiontree** [comment]
The issue for us was that we were working in a monorepo and the extension was not resolving the tailwind config properly from the project root, so no custom tailwind options were being recognized by the extension. We resolved it by updating the `eslint.workingDirectories` setting to point to each app/package, eg:

```
"eslint.workingDirectories": [{ "pattern": "./packages/*/" }, { "pattern": "./apps/*/" }],
```

**Comment by dbaeumer** [comment]
@joel-saupe-visiontree thanks for the advice.

@Nurai1 I closed the issue because it needs more information and has not had recent activity. See also our issue reporting guidelines.

Happy Coding!

**Comment by ht-lovrozagar** [comment]
For monorepo, targeting the config file from the very root works without the need to change vscode settings.json.

Add the following to a specific app's eslint configuration:

```
  "settings": {
    "tailwindcss": {
      "config": "apps/app-1/tailwind.config.js"
    }
  }
```

**Comment by Tordek** [comment]
I ran into this issue in a monorepo too and my fix was:

```js
  "settings": {
    "tailwindcss": {
      "config": require.resolve("tailwind.config.js")
    }
  }
```

**Comment by ben-xD** [comment]
I guess @ht-lovrozagar's answer assumes your project only has 1 app/package that uses tailwind. There's only 1 string you can pass in `settings.tailwindcss.config`. So I believe VSCode configuration is much **more versatile**: https://github.com/microsoft/vscode-eslint/issues/1706#issuecomment-1785746444.
